src/tqec/templates/indices/qubit.py

```diff
@@ -70,22 +70,22 @@ def expected_plaquettes_number(self) -> int:
         return 14
 
 
-class Qubit4WayJunctionTemplate(RectangularTemplate):
+class QubitSpatialJunctionTemplate(RectangularTemplate):
```

Comment:
Initially it's kind of surprising to me that this template would capture all the spatial junction cases as there is some asymmetry in the template, e.g. `17` is symmetry around `5`(`8`) at the top-left(bottom-right) corner while `13`/`14`(`15`/`16`) is not at the top-right(bottom-left) corner. Then I realized that it's because `5` and `7` are of different basis and the spatial junction only needs to handle hook errors of one basis.

Reply:
The current implementation might not cover all the cases, but I think it can. The only way to know for sure is by implementing the cases and see if that is possible. This should be tested in the different RPNGTemplate instantiations in the library.